Streetcode/Streetcode.BLL/Services/Authentication/TokenService.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Data;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Localization;
+using Microsoft.IdentityModel.Tokens;
+using Streetcode.BLL.Interfaces.Authentication;
+using Streetcode.DAL.Entities.Users;
+using Streetcode.DAL.Persistence;
+using Streetcode.BLL.Utils.Options;
+
+namespace Streetcode.BLL.Services.Authentication
+{
+    public sealed class TokenService : ITokenService
+    {
+        private readonly SigningCredentials _signingCredentials;
+        private readonly JwtOptions _jwtOptions;
+        private readonly StreetcodeDbContext _dbContext;
+        private readonly JwtSecurityTokenHandler _jwtSecurityTokenHandler;
+
+        public TokenService(IConfiguration configuration, StreetcodeDbContext dbContext)
+        {
+            var c = configuration
+              .GetSection("Jwt");
+            _jwtOptions = configuration
+              .GetSection("Jwt")
+              .Get<JwtOptions>() !;
+
+            _dbContext = dbContext;
+
+            var symmetricSecurityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_jwtOptions.Key));
+            _signingCredentials = new SigningCredentials(symmetricSecurityKey, SecurityAlgorithms.HmacSha256);
+            _jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
+        }
+
+        public JwtSecurityToken GenerateJWTToken(User? user)
+        {
+            if (user is null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            var userRoleId = _dbContext.UserRoles
+                .AsNoTracking()
+                .Where(userRole => userRole.UserId == user.Id)
+                .Select(userRole => userRole.RoleId)
+                .FirstOrDefault();
```

Comment:
I think you don't need .FirstOrDefault here. You can read more about IQuarable if you want